packages/python/plotly/plotly/express/_core.py

```diff
@@ -1028,26 +1052,27 @@ def _get_reserved_col_names(args):
                 continue
             elif isinstance(arg, str):  # no need to add ints since kw arg are not ints
                 reserved_names.add(arg)
-            elif isinstance(arg, pd.Series):
-                arg_name = arg.name
-                if arg_name and hasattr(df, arg_name):
-                    in_df = arg is df[arg_name]
+            elif nw.dependencies.is_into_series(arg):
+                arg_series = nw.from_native(arg, series_only=True)
+                arg_name = arg_series.name
+                if arg_name and arg_name in df.columns:
+                    in_df = (arg_series == df.get_column(arg_name)).all()
                     if in_df:
                         reserved_names.add(arg_name)
-            elif arg is df.index and arg.name is not None:
+            elif arg is nw.maybe_get_index(df) and arg.name is not None:
                 reserved_names.add(arg.name)
 
     return reserved_names
 
 
-def _is_col_list(columns, arg):
+def _is_col_list(columns, arg, is_pd_like, native_namespace):
     """Returns True if arg looks like it's a list of columns or references to columns
     in df_input, and False otherwise (in which case it's assumed to be a single column
     or reference to a column).
     """
     if arg is None or isinstance(arg, str) or isinstance(arg, int):
         return False
-    if isinstance(arg, pd.MultiIndex):
+    if is_pd_like and isinstance(arg, native_namespace.MultiIndex):
```

Comment:
Couldn't you simply do `native_namespace.__name__ == "pandas"` and drop the need for a separate `is_pd_like` argument? Or does that feel too awkward?

Reply:
`is_pd_like` is not just pandas, we support `pandas`, `modin.pandas` and `cudf`, that's the reason for a slightly more generic path